autogluon/task/text_prediction/models/basic_v1.py

```diff
@@ -0,0 +1,911 @@
+    cfg.freeze()
+    return cfg
+
+
+def electra_base():
+    """The search space of Electra Base"""
+    cfg = base_cfg()
+    cfg.defrost()
+    cfg.optimization.layerwise_lr_decay = 0.8
+    cfg.freeze()
+    return cfg
+
+
+def mobile_bert():
+    """The search space of MobileBERT"""
+    cfg = base_cfg()
+    cfg.defrost()
+    cfg.optimization.layerwise_lr_decay = -1
+    cfg.model.backbone.name = 'google_uncased_mobilebert'
+    cfg.optimization.lr = 1E-5
+    cfg.optimization.num_train_epochs = 5.0
+    cfg.freeze()
+    return cfg
+
+
+def calculate_metric_scores(metrics, predictions, gt_labels,
+                            pos_label=1):
+    """
+
+    Parameters
+    ----------
+    metrics
+        A list of metric names
+    predictions
+    gt_labels
+    pos_label
+
+    Returns
+    -------
+    metric_scores
+        A dictionary contains key --> metric scores
+    """
+    if isinstance(metrics, str):
+        metrics = [metrics]
+    metric_scores = collections.OrderedDict()
+    for metric_name in metrics:
+        if metric_name == 'acc':
+            metric_scores[metric_name] = float(accuracy_score(gt_labels,
+                                                              predictions.argmax(axis=-1)))
+        elif metric_name == 'f1':
+            metric_scores[metric_name] = float(f1_score(gt_labels,
+                                                        predictions.argmax(axis=-1),
+                                                        pos_label=pos_label))
+        elif metric_name == 'mcc':
+            metric_scores[metric_name] = float(matthews_corrcoef(gt_labels,
+                                                                 predictions.argmax(axis=-1)))
+        elif metric_name == 'auc':
+            metric_scores[metric_name] = float(roc_auc_score(gt_labels, predictions[:, pos_label]))
+        elif metric_name == 'nll':
+            metric_scores[metric_name]\
+                = float(- np.log(predictions[np.arange(gt_labels.shape[0]), gt_labels]).mean())
+        elif metric_name == 'pearsonr':
+            metric_scores[metric_name] = float(pearsonr(gt_labels, predictions)[0])
+        elif metric_name == 'spearmanr':
+            metric_scores[metric_name] = float(spearmanr(gt_labels, predictions)[0])
+        elif metric_name == 'mse':
+            metric_scores[metric_name] = float(np.square(predictions - gt_labels).mean())
+        elif metric_name == 'rmse':
+            metric_scores[metric_name] = float(np.sqrt(np.square(predictions - gt_labels).mean()))
+        elif metric_name == 'mae':
+            metric_scores[metric_name] = float(np.abs(predictions - gt_labels).mean())
+        else:
+            raise ValueError('Unknown metric = {}'.format(metric_name))
+    metric_scores = collections.OrderedDict(
+        [(k, v.item() if isinstance(v, np.ndarray) else v)
+         for k, v in metric_scores.items()])
+    return metric_scores
+
+
+def is_better_score(metric_name, baseline, new_score):
```

Comment:
What about custom metrics? Perhaps this should be inherent in the metric function/class itself rather than hardcoded here. See Tabular for example.